### PR TITLE
Adds generic SMTP driver

### LINKED ISSUE #508: PHP 8.x Support
Hi @TangRufus,
I use this package for nearly every project, it is so useful and easy to set-up. 
However I am not sure whether you continue to update it. E.g. by adding PHP 8 as requirement in composer.json.
I also added a PR quite a while ago: https://github.com/ItinerisLtd/wp-phpmailer/pull/486

Kind regards,

Philipp

### PR DESCRIPTION
**Pull request by ouun**
This is such a useful and streight forward SMTP implementation. Thanks a lot for the package.
However I stumpled over it again and wondered another time, why there is no simple SMTP driver one can use with any SMTP compatible service. So I added one for my own and I am sure that this will be useful for others, too.

Thanks and kind regards,

Philipp

Closes #508

### CODE REVIEW

**Comment by codepuncher** [comment]
@ouun thanks for the PR.

Could you look into resolving the failed checks before we can look to merge this?

**Comment by ouun** [comment]
@codepuncher thanks for getting back on this. All checks succeed now, so would be great to get this merged.
Thanks! :)

File: src/ConfigFactory.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 namespace Itineris\WPPHPMailer;
 
+use Itineris\WPPHPMailer\Drivers\SMTPDriver;
 use Itineris\WPPHPMailer\Drivers\MailHogDriver;
 use Itineris\WPPHPMailer\Drivers\MailtrapDriver;
 use Itineris\WPPHPMailer\Drivers\Office365Driver;
```
Comment (codepuncher) [inline src/ConfigFactory.php:10]:
```suggestion
use Itineris\WPPHPMailer\Drivers\MailHogDriver;
use Itineris\WPPHPMailer\Drivers\MailtrapDriver;
use Itineris\WPPHPMailer\Drivers\Office365Driver;
use Itineris\WPPHPMailer\Drivers\SMTPDriver;
```

File: tests/unit/ConfigFactoryTest.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 namespace Itineris\WPPHPMailer;
 
 use Codeception\Test\Unit;
+use Itineris\WPPHPMailer\Drivers\SMTPDriver;
```
Comment (codepuncher) [inline tests/unit/ConfigFactoryTest.php:8]: Sort lines alphabetically

File: tests/unit/ConfigFactoryTest.php [hunk 3]
```diff
@@ -29,13 +30,14 @@ public function testThrowDriverNotFoundException(): void
             ->once();
 
         WP_Mock::expectFilter('wp_phpmailer_drivers', [
+            'smtp' => SMTPDriver::class,
```
Comment (codepuncher) [inline tests/unit/ConfigFactoryTest.php:33]: Sort lines alphabetically

**Review by codepuncher — COMMENTED** [review]
Thanks for sorting that @ouun. Almost ready to merge.
Final thing, could you resort the imports and uses so we preserve the alphabetical order?
Once that's done we can merge it :slightly_smiling_face:

**Comment by ouun** [comment]
@codepuncher uses and imports are sorted now. Cheers.

**Review by codepuncher — APPROVED** [review]

**Comment by codepuncher** [comment]
Lovely; thanks for that @ouun and apologies it took so long for me to get to this!

**Pull request merged**